diff --git a/pyomo/contrib/pynumero/src/CMakeLists.txt b/pyomo/contrib/pynumero/src/CMakeLists.txt
index 001e131..b13bc6a 100644
--- a/pyomo/contrib/pynumero/src/CMakeLists.txt
+++ b/pyomo/contrib/pynumero/src/CMakeLists.txt
@@ -11,6 +11,7 @@ include(ExternalProject)
 OPTION(BUILD_ASL "Build the PyNumero ASL interface" ON)
 OPTION(BUILD_MA27 "Build the PyNumero ma27 interface" OFF)
 OPTION(BUILD_MA57 "Build the PyNumero ma57 interface" OFF)
+OPTION(ENABLE_HSL "Enable the HSL library interfaces" ON)

 # Dependencies that we manage / can install
 SET(AMPLMP_TAG "3.1.0" CACHE STRING
@@ -98,6 +99,12 @@ IF( MA27_LIBRARY OR MA27_OBJECT )
     set_property(CACHE BUILD_MA27 PROPERTY VALUE ON)
 ENDIF()

+#...but if the HSL interface is not enabled, do not build the MA* libraries
+IF( NOT ENABLE_HSL )
+    set_property(CACHE BUILD_MA27 PROPERTY VALUE OFF)
+    set_property(CACHE BUILD_MA57 PROPERTY VALUE OFF)
+ENDIF()
+
 # If BUILD_AMPLMP_IF_NEEDED is set and we couldn't find / weren't
 # pointed to an ASL build, then we will forcibly enable the AMPLMP build
 # to provide the ASL.
